pkg/agent/flowexporter/denyconnections/deny_connections.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package denyconnections
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/flowexporter"
+)
+
+type DenyConnectionStore interface {
+	// AddOrUpdateConnection adds or updates connection into deny connection map.
+	AddOrUpdateConnection(conn *flowexporter.DenyConnection)
+	// ForAllConnectionsDo executes the callback for each connection in deny connection map.
+	ForAllConnectionsDo(callback flowexporter.DenyConnectionMapCallBack) error
+	// ContainsConnection checks whether connection exists in deny connection store.
+	ContainsConnection(flowKey flowexporter.Tuple) bool
+	// DeleteConnWithoutLock deletes the deny connection from the connection map given
+	// the connection key without grabbing the lock. Caller is expected to grab lock.
+	DeleteConnWithoutLock(connKey flowexporter.ConnectionKey) error
+	// GetNumOfConnections returns number of connections in deny connection map.
+	GetNumOfConnections() int
+}
```

Comment:
I have mentioned this several times already in this review, but is there no way to unify the connection stores for "regular" connections & denied connections?

Would it make sense to have:
- a base struct shared by both connection types
- have the connection store methods use some `interface` type as the stored object

If there are some subtle differences, please let me know. We shouldn't make the code more complex for the sake of unification, but it seems like there is a missed opportunity here. I'll let @srikartati chime in as well.

Reply:
Yes, a base structure and embedding it in both regular and denied connection type makes sense and can consolidate code.
Having one interface for connection store and different structs for regular and denied connection store implementing the interface also sounds good. This may involve some refactoring in tests. Will let @zyiou comment more on this feasibility.

Reply:
Yes, it makes sense to me to use base structure with common methods. will do the refactoring accordingly. Thanks!